db-service/lib/search.js

```diff
@@ -94,13 +94,18 @@ const _getSearchableColumns = entity => {
     if (annotatedColumnValue) return true
 
     // calculated elements are only searchable if requested through `@cds.search` 
-    if(column.value) return false
+    if (column.value) return false
 
     // if at least one element is explicitly annotated as searchable, e.g.:
     // `@cds.search { element1: true }` or `@cds.search { element1 }`
     // and it is not the current column name, then it must be excluded from the search
     if (atLeastOneColumnIsSearchable) return false
 
+    if (element?.['@Common.Text']?.['='] && element?.isAssociation) {
+      deepSearchCandidates.push({ ref: element['@Common.Text']['='].split('.') })
+      return false
+    } 
+
```

Comment:
We should move that out to a model pre-processor, that runs once on server start instead of polluting our db layers with such knowledge of OData / Fiori concepts , and doing that again and again per query at runtime